### PR TITLE
Update for building docker image 

### PR DESCRIPTION
**Pull request by merlintang**
### what is proposed in this PR?
update the docker build file for go model dependency 

### how to test this PR? 
docker build can work

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost-operator/29)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+ENV GO111MODULE=on
 
 # Copy in the go src
 WORKDIR /go/src/github.com/kubeflow/xgboost-operator
+
+COPY go.mod .
+COPY go.sum .
+# COPY vendor/ vendor/
+
+RUN go mod download
```
Comment (terrytangyuan) [inline Dockerfile]: What's the purpose of this change? Is this being used somewhere?

Comment (merlintang) [reply Dockerfile]: we need to setup the go build env based on the go.sum and go.mode. or else, the docker build would break.

**Comment by merlintang** [comment]
@terrytangyuan I have updated the pr. thanks

**Comment by merlintang** [comment]
thanks
